### ISSUE TITLE
NA/1.0/0.0 results for SAIGE-GENE+ when MAC = 10

### ISSUE THREAD

**Issue by dvg-p4** [saigegit/SAIGE#60]
There appears to be a bug in region-based analyses for regions that have a MAC of precisely 10. In one analysis I did (unfortunately proprietary data so I can't share the particulars), all the genes with MAC = 10 returned a p-value of either 0.0, 1.0, or NA. No MAC count other than 10 led to 0.0 or NA p-values. The ones that had a 0.0 P-value also returned nonsensically high beta values, >10^20, despite prior versions of SAIGE (0.44.2) returning nonsignificant p-values and betas near zero for these genes.

I suspect this may be a bug related to the new minmum threshold for categorical variance ratios and maximum threshold for collapsing ultra-rare variants. Both of these seem to be set to 10.0 by default, so it seems that something goes wrong with genes precisely on the threshold. Perhaps setting these to 10.5 will avoid the issue--I will do a test run with that parameter and report back.

**Comment by dvg-p4** [comment]
Manually setting the thresholds to 10.5 appears to fix the issue. With:
```
--MACCutoff_to_CollapseUltraRare=10.5
--cateVarRatioMinMACVecExclude=10.5,20.5
```
for step 2, I get a reasonable-looking distribution of p-values and betas---no more 0.0s, 1.0s, or NAs.

Do you think that `--cateVarRatioMinMACVecExclude` should also be changed from `10,20.5` to `10.5,20.5` for step 1?

**Comment by dvg-p4** [comment]
I should clarify, I was running both step 1 and 2 within the 1.1.6.1 container from docker hub, on the shell command line with [step1_fitNULLGLMM.R](https://github.com/saigegit/SAIGE/blob/main/extdata/step1_fitNULLGLMM.R) and [step2_SPAtests.R](https://github.com/saigegit/SAIGE/blob/main/extdata/step2_SPAtests.R)

**Comment by weizhou0** [comment]
Hi @dvg-p4, 

This issue should have been fixed in 1.1.6.3. --cateVarRatioMinMACVecExclude in both steps should be consistent. 

Thanks,
Wei
